### PR TITLE
Start test reorg

### PR DESCRIPTION
**Pull request by tmontaigu**
This starts splitting the long test radix tests files into
smaller ones, starting with the add family of function.

---

This adds degrees and noise levels checks as well as comparing
individual decrypted block values with their degrees.

---

smart_add test is not 'hardened' yet, but I want some feedback already

### CODE REVIEW

**Comment by tmontaigu** [comment]
Seems like it caught some problems in GPU

**Comment by tmontaigu** [comment]
Its simple the noise level is not updated after carry propagation

**Comment by IceTDrinker** [comment]
> Its simple the noise level is not updated after carry propagation

Do we have a fix for that ?

**Comment by tmontaigu** [comment]
setting the noise level to nominal (after checking the code ends with a pbs)

File: tfhe/src/integer/server_key/radix_parallel/tests_unsigned/test_add.rs [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+fn integer_default_overflowing_add<P>(param: P)
+where
+    P: Into<PBSParameters>,
+{
+    let executor = CpuFunctionExecutor::new(&ServerKey::unsigned_overflowing_add_parallelized);
+    default_overflowing_add_test(param, executor);
+}
+
+impl ExpectedNoiseLevels {
+    fn after_add(&mut self, lhs: &RadixCiphertext, rhs: &RadixCiphertext) -> &Self {
```
Comment (IceTDrinker) [inline tfhe/src/integer/server_key/radix_parallel/tests_unsigned/test_add.rs:89]: maybe after_unchecked_add ?

File: tfhe/src/integer/server_key/radix_parallel/tests_unsigned/test_add.rs [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+                .iter()
+                .zip(rhs.blocks.iter())
+                .map(|(a, b)| a.noise_level + b.noise_level),
+        );
+        self
+    }
+}
+
+impl ExpectedDegrees {
+    fn after_add(&mut self, lhs: &RadixCiphertext, rhs: &RadixCiphertext) -> &Self {
```
Comment (IceTDrinker) [inline tfhe/src/integer/server_key/radix_parallel/tests_unsigned/test_add.rs:101]: same here

File: tfhe/src/integer/server_key/radix_parallel/tests_unsigned/mod.rs [hunk 3]
```diff
@@ -1,30 +1,251 @@
...
+        assert_eq!(self.values.len(), ct.blocks.len());
+        for (i, (block, expected_noise)) in ct
+            .blocks
+            .iter()
+            .zip(self.values.iter().copied())
+            .enumerate()
+        {
+            assert_eq!(
+                block.noise_level, expected_noise,
+                "Block at index {i} have noise level {:?}, but {expected_noise:?} was expected",
```
Comment (IceTDrinker) [inline tfhe/src/integer/server_key/radix_parallel/tests_unsigned/mod.rs:222]: have -> has

File: tfhe/src/integer/server_key/radix_parallel/tests_unsigned/mod.rs [hunk 4]
```diff
@@ -1,30 +1,251 @@
...
+        assert_eq!(self.values.len(), ct.blocks.len());
+        for (i, (block, expected_degree)) in ct
+            .blocks
+            .iter()
+            .zip(self.values.iter().copied())
+            .enumerate()
+        {
+            assert_eq!(
+                block.degree, expected_degree,
+                "Block at index {i} have degree {:?}, but {expected_degree:?} was expected",
```
Comment (IceTDrinker) [inline tfhe/src/integer/server_key/radix_parallel/tests_unsigned/mod.rs:241]: have -> has

**Review by IceTDrinker — COMMENTED** [review]
2 nitpicks let me know if you want to update, otherwise tell me and I approve !

This is a very nice step forward for testing, I like the hardening and the new primitives to test ! 🙂 Especially the per block decryption you have been talking about before

**Review by IceTDrinker — APPROVED** [review]
Thanks a lot !

**Pull request merged**